addons/binding/org.openhab.binding.bosesoundtouch10/src/main/java/org/openhab/binding/bosesoundtouch10/handler/BoseSoundTouch10Handler.java

```diff
@@ -0,0 +1,378 @@
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.openhab.binding.bosesoundtouch10.BoseSoundTouch10BindingConstants;
+import org.openhab.binding.bosesoundtouch10.helper.SimpleCallBackInterface;
+import org.openhab.binding.bosesoundtouch10.helper.SimpleSocketListener;
+import org.openhab.binding.bosesoundtouch10.helper.WSHelper;
+import org.openhab.binding.bosesoundtouch10.helper.WSHelperInterface;
+import org.openhab.binding.bosesoundtouch10.helper.XmlHelper;
+import org.openhab.binding.bosesoundtouch10.helper.XmlResult;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link BoseSoundTouch10Handler} is responsible for handling commands, which are
+ * sent to one of the channels, and handling messages coming from the speaker.
+ *
+ * @author syracom - Initial contribution
+ */
+public class BoseSoundTouch10Handler extends BaseThingHandler implements SimpleCallBackInterface {
+
+    private Logger logger = LoggerFactory.getLogger(BoseSoundTouch10Handler.class);
+    private String soundTouchURL;
+    private WebSocketClient client;
+    private WebSocketListener socketListener;
+    private WSHelperInterface wsHelper;
+    private XmlHelper xmlHelper;
+
+    public enum KeyState {
+        PRESS("press"),
+        RELEASE("release");
+
+        private String value;
+
+        private KeyState(String value) {
+            this.value = value;
+        }
+
+        public String getValue() {
+            return value;
+        }
+    }
+
+    public enum Sources {
+        STANDBY,
+        AUX,
+        BLUETOOTH,
+        INTERNET_RADIO
+    }
+
+    public enum BSTKeys {
+        PLAY,
+        PAUSE,
+        POWER,
+        PRESET_1,
+        PRESET_2,
+        PRESET_3,
+        PRESET_4,
+        PRESET_5,
+        PRESET_6,
+    }
+
+    public BoseSoundTouch10Handler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        String response = "";
+        checkForReInit();
+        switch (channelUID.getId()) {
+            case BoseSoundTouch10BindingConstants.CHANNEL_POWER:
+                if (command instanceof OnOffType) {
+                    response = wsHelper.pressAndReleaseButtonOnSpeaker(BSTKeys.POWER);
+                }
+                break;
+            case BoseSoundTouch10BindingConstants.CHANNEL_VOLUME:
+                if (command instanceof RefreshType) {
+                    response = wsHelper.get("/volume");
+                    if (response.equals("error")) {
+                        break;
+                    }
+                    String volume = xmlHelper.parsePath("/volume/actualvolume", response);
```

Comment:
The XmlHelper class already has a few Xpath constants. Could you put this in there as well?